code/API_definitions/connectivity-insights-subscriptions.yaml

```
@@ -0,0 +1,1265 @@
+          $ref: "#/components/schemas/Source"
+        type:
+          $ref: "#/components/schemas/Event-typeNotification"
+        specversion:
+          type: string
+          description: |
+            Version of the specification to which this event conforms (must be
+             1.0 if it conforms to cloudevents 1.0.2 version)
+          enum:
+            - "1.0"
+        datacontenttype:
+          type: string
+          description: |
+           'media-type that describes the event payload encoding,
+           must be "application/json" for CAMARA APIs'
+          enum:
+            - application/json
+        data:
+          $ref: "#/components/schemas/NetworkQualityInsight"
+        time:
+          $ref: "#/components/schemas/DateTime"
+      discriminator:
+        propertyName: "type"
+        mapping:
+          org.camaraproject.connectivityinsights.v0.networkQuality-event:
+            "#/components/schemas/NetworkQualityTypeEvent"
+          org.camaraproject.connectivityinsights.v0.eventSubscriptionEnds:
+            "#/components/schemas/EventSubscriptionEnds"
+
+    NetworkQualityTypeEvent:
+      description: event structure for event-type event
+      allOf:
+        - $ref: "#/components/schemas/CloudEvent"
+        - type: object
+          properties:
+            data:
+              $ref: "#/components/schemas/NetworkQualityInsight"
+
+    EventSubscriptionEnds:
+      description: event structure for event subscription ends
+      allOf:
+        - $ref: "#/components/schemas/CloudEvent"
+        - type: object
+          properties:
+            data:
+              $ref: "#/components/schemas/SubscriptionEnds"
+
+    SubscriptionEnds:
+      description: Event detail structure for SUBSCRIPTION_ENDS event
+      type: object
+      required:
+        - terminationReason
+        - subscriptionId
+      properties:
+        terminationReason:
+          $ref: "#/components/schemas/TerminationReason"
+        subscriptionId:
+          $ref: "#/components/schemas/SubscriptionId"
+        terminationDescription:
+          type: string
+
+    TerminationReason:
+      type: string
+      description: |
+        - NETWORK_TERMINATED - API server stopped sending notification
+        - SUBSCRIPTION_EXPIRED - Subscription expire time (optionally set by
+        the requester) has been reached
+        - MAX_EVENTS_REACHED - Maximum number of events (optionally set by the
+        requester) has been reached
+        - ACCESS_TOKEN_EXPIRED - Access Token sinkCredential (optionally set by
+        the requester) expiration time has been reached
+      enum:
+        - MAX_EVENTS_REACHED
+        - NETWORK_TERMINATED
+        - SUBSCRIPTION_EXPIRED
+        - ACCESS_TOKEN_EXPIRED
+
+    DateTime:
+      type: string
+      format: date-time
+      description: |
+        Timestamp of when the occurrence happened. Must adhere to RFC 3339.
+      example: "2018-04-05T17:31:00Z"
+
+    Event-typeNotification:
```

This was changed in Commonalities 0.4.0-rc.2

```suggestion
    EventTypeNotification:
```